### ISSUE TITLE
Blocker bug - plugin explodes if you already have a tooltip property anywhere in your app or any of your apps components

### ISSUE THREAD

**Issue by coladarci** [sir-dunxalot/ember-tooltips#47]
This plugin adds a `destroyTooltip: on('willDestroyElement', function () {` method that is in the mixin that gets injected everywhere.

Inside there it says:

```
var tooltip = this.get('tooltip');

if (tooltip) {
  .... DO SOME CRAY STUFF THAT ONLY MY TOOLTIP RESPONDS TO
}
```

So, if you have a tooltip property, or worse yet, you are using someone else's add-on who has a tooltip property, your app will come to a screeching halt when you navigate away from that view.

The fix should be that this plugin uses a more specific name for that property that is less likely to conflict AND checks to see if the `tooltip` it gets back is, in fact, the type of tooltip you are expecting.

**Comment by sir-dunxalot** [comment]
I think the second part of that (checking the type/constructor of `tooltip`) is reasonable. The first part, about renaming the `tooltip` property seems like more of an odd edge case. My reasoning being that this is the tooltip library and, thus, should probably own the tooltip property. I get there are some situations that will overlap (e.g. using a UI library that has tooltips built in) but let's implement the second part of your fix and reassess if it continues to be a pain. How does that sound?

**Comment by coladarci** [comment]
To my knowledge, checking the constructor in JS is a dogey area - I have a simpler fix in a local branch - https://github.com/sir-dunxalot/ember-tooltips/pull/48 - pretty "hacky" but our prod site went down so I needed something quick :) Thoughts?

**Comment by coladarci** [comment]
Another way I've handled this in the past is to add a property to your tooltip object, like isEmberTooltip, so you can replace this conditional w/ a check for that property. Still hacky but a little more explicit.. Thoughts?

**Comment by sir-dunxalot** [comment]
I think either of those should work. The underlying library has a `Tooltip` class so I think we'd be ok there. But I defer to you if you know of any other issues checking the constructor. I'll test it out and if that doesn't work we'll use the second approach - I've used it in multiple projects before and it's been reliable.

**Comment by coladarci** [comment]
Hey @sir-dunxalot - have you had a chance to check this out? Thanks!

**Comment by sir-dunxalot** [comment]
Sorry, not yet. I'll try to do so before the weekend. Thanks for the reminder.

**Comment by sir-dunxalot** [comment]
This should be fixed by #48, which is now on master and released. If you're still having issues we can reopen this issue.
